src/lib/bootloader/systemdboot_widgets.rb

```diff
@@ -0,0 +1,169 @@
+
+      def label
+        _("&Timeout in Seconds")
+      end
+
+      def help
+        _("<p><b>Timeout in Seconds</b>\n" \
+          "specifies the time the boot loader will wait until the default kernel is loaded.</p>\n")
+      end
+
+      def init
+        self.value = systemdboot.menue_timeout.to_i
+      end
+
+      def store
+        systemdboot.menue_timeout = value.to_s
+      end
+    end
+
+    # Represents switcher for secure boot on EFI
+    class SecureBootWidget < CWM::CheckBox
+      include SystemdBootHelper
+
+      def initialize
+        textdomain "bootloader"
+
+        super
+      end
+
+      def label
+        _("&Secure Boot Support")
+      end
+
+      def help
+        _(
+          "<p><b>Secure Boot Support</b> if checked enables Secure Boot support.<br>" \
+          "This does not turn on secure booting. " \
+          "It only sets up the boot loader in a way that supports secure booting. " \
+          "You still have to enable Secure Boot in the UEFI Firmware.</p> "
+        )
+      end
+
+      def init
+        self.value = systemdboot.secure_boot
+      end
+
+      def store
+        systemdboot.secure_boot = value
+      end
+    end
+
+    # represents Tab with kernel related configuration
+    class KernelTab < CWM::Tab
+      def label
+        textdomain "bootloader"
+
+        _("&Kernel Parameters")
+      end
+
+      def contents
+        VBox(
+          VSpacing(1),
+          HBox(
+            HSpacing(1),
+            HStretch()
+          ),
+          VStretch()
+        )
+      end
+    end
+
+    # Represent tab with options related to stage1 location and bootloader type
+    class BootCodeTab < CWM::Tab
+      include SystemdBootHelper
+
+      def label
+        textdomain "bootloader"
+
+        _("Boot Co&de Options")
+      end
+
+      def contents
+        VBox(
+          VSpacing(1),
+          HBox(
+            HSpacing(1),
+            Left(LoaderTypeWidget.new)
+          ),
+          VSpacing(1),
+          *widgets,
+          VStretch()
+        )
+      end
+
+    private
+
+      def widgets
+        w = []
+        w << SecureBootWidget.new if secure_boot_widget?
+
+        w.map do |widget|
+          MarginBox(horizontal_margin, 0, Left(widget))
+        end
+      end
+
+      def horizontal_margin
+        @horizontal_margin ||= Yast::UI.TextMode ? 1 : 1.5
+      end
+
+      def secure_boot_widget?
+        Systeminfo.secure_boot_available?(systemdboot.name)
+      end
+    end
+
+    # Represents bootloader specific options like its timeout,
+    # default section or password protection
+    class BootloaderTab < CWM::Tab
+      def label
+        textdomain "bootloader"
+
+        _("Boot&loader Options")
+      end
+
+      def contents
+        VBox(
+          VSpacing(2),
+          HBox(
+            HSpacing(1),
+            TimeoutWidget.new,
+            HSpacing(1)
+          ),
+          VStretch()
+        )
+      end
+    end
+  end
+end
```

Comment:
good as POC, but I think if more settings like kernel params or cpu mitigation appears we should somehow abstract widgets to be able to reuse ones we have for grub2 as they looks almost identical.